src/lib.rs

```diff
@@ -572,48 +562,32 @@ impl<K: EnrKey> Enr<K> {
 
     /// Sets the `udp` field of the ENR. Returns any pre-existing UDP port in the record.
     pub fn set_udp4(&mut self, udp: u16, key: &K) -> Result<Option<u16>, EnrError> {
-        if let Some(udp_bytes) = self.insert("udp", &udp.to_be_bytes(), key)? {
-            if udp_bytes.len() <= 2 {
-                let mut v = [0_u8; 2];
-                v[2 - udp_bytes.len()..].copy_from_slice(&udp_bytes);
-                return Ok(Some(u16::from_be_bytes(v)));
-            }
+        if let Some(udp_bytes) = self.insert_raw_rlp("udp", rlp::encode(&udp).freeze(), key)? {
+            return Ok(rlp::decode(&udp_bytes).ok());
```

Comment:
if we sanitize correctly, this should never fail right? (referring to the `ok()`)

Reply:
I would still be careful here with putting an `expect()` in however. If for some reason someone managed to craft some crazy ENR that passes our decoding, we probably don't want to crash whatever node uses it.